inject.js

```diff
@@ -459,23 +502,26 @@
 
   function runAction(action, document, value, e) {
     if (tc.settings.audioBoolean) {
-      var mediaTags = document.querySelectorAll('video,audio');
+      var mediaTags = getShadow(document.body).filter(x => {
+        return x.tagName == 'AUDIO' || x.tagName == 'VIDEO'
+      });
     } else {
-      var mediaTags = document.querySelectorAll('video');
+      var mediaTags = getShadow(document.body).filter(x => x.tagName == 'VIDEO');
     }
 
     mediaTags.forEach = Array.prototype.forEach;
 
     // Get the controller that was used if called from a button press event e
     if (e) {
       var targetController = e.target.getRootNode().host;
-    } 
+    }
 
     mediaTags.forEach(function(v) {
       var id = v.dataset['vscid'];
-      var controller = document.querySelector(`div[data-vscid="${id}"]`);
-
-      // Don't change video speed if the video has a different controller	
+      var controller = getShadow(document.body).filter(x => {
```

Comment:
We duplicate the same call here and on line 183, can we extract this into a reusable function?

Reply:
ok.

Reply:
i did and now that function is called getController